helm-chart/charts/cb-dragonfly/values.yaml

```
@@ -70,16 +71,22 @@ kafka:
     ADVERTISED_PORT: 32000
     #    advertised.listeners: PLAINTEXT://localhost:9092
     DELETE_TOPIC_ENABLE: "true"
-    LOG_RETENTION_MS: 10000
-    LOG_RETENTION_CHECK_INTERVAL_MS: 5000
+    LOG_RETENTION_HOURS: 1
+    # 1 hours (delete logs)
+    LOG_RETENTION_BYTES: 524288000
+    # 500 MB
+    LOG_SEGMENT_BYTES: 262144000
+    # 250 MB
```

이런 에러가 나네요..

```
[2021-09-28 11:31:30,061] ERROR Exiting Kafka due to fatal exception (kafka.Kafka$)
org.apache.kafka.common.config.ConfigException: Invalid value 2.62144e+08 for configuration log.segment.bytes: Not a number of type INT
        at org.apache.kafka.common.config.ConfigDef.parseType(ConfigDef.java:726)
        at org.apache.kafka.common.config.ConfigDef.parseValue(ConfigDef.java:474)
        at org.apache.kafka.common.config.ConfigDef.parse(ConfigDef.java:467)
        at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:108)
        at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:142)
        at kafka.server.KafkaConfig.<init>(KafkaConfig.scala:1119)
        at kafka.server.KafkaConfig.<init>(KafkaConfig.scala:1122)
        at kafka.server.KafkaConfig$.fromProps(KafkaConfig.scala:1102)
        at kafka.server.KafkaServerStartable$.fromProps(KafkaServerStartable.scala:34)
        at kafka.server.KafkaServerStartable$.fromProps(KafkaServerStartable.scala:29)
        at kafka.Kafka$.main(Kafka.scala:68)
        at kafka.Kafka.main(Kafka.scala)
```

---

https://stackoverflow.com/questions/48871262/kafka-broker-throwing-configexception-invalid-value-configuration-log-segment-by

이 글을 보고 확인해 보니
262144000 은 `Int.MaxInt` = 2147483647 보다 작아서
Int range 문제는 아니었고..

---

```suggestion
    LOG_SEGMENT_BYTES: "262144000"
    # 250 MB
```

이렇게 하니까 kafka가 정상적으로 실행됩니다.. ㄷㄷ

숫자를 따옴표로 감싸지 않으면
Helm chart `values.yaml` 의 `configurationOverrides` 를 통해 ENV가 넘어가는 과정에서 "`2.62144e+08`" 로 넘어가고
숫자를 따옴표로 감싸면 "`262144000`" 으로 넘어가나 봅니다.. ㄷㄷ